core/src/main/java/org/opensearch/sql/storage/StorageEngine.java

```diff
@@ -30,4 +32,8 @@ default Collection<FunctionResolver> getFunctions() {
     return Collections.emptyList();
   }
 
+  default TableScanOperator getTableScan(String indexName, String scrollId) {
```

Comment:
why get TableScanOperator in StorageEngine?

Reply:
It is used in cursor deserialization/parsing, this happens in `:core` module. All implementation of `StorageEngine` are defined outside of `:core` and unavailable and that point.

Reply:
@penghuo does this design make sense? Would love to know what you think.